lib/vector-core/src/buffers/disk/leveldb_buffer.rs

```diff
@@ -411,7 +358,7 @@ impl super::DiskBuffer for Buffer {
             tail = if iter.valid() { iter.key().0 + 1 } else { 0 };
         }
 
-        let current_size = Arc::new(AtomicUsize::new(initial_size));
+        let current_size = Arc::new(AtomicUsize::new(0));
```

Comment:
I think calling this `current_size` is a bit misleading now. Could we rename this variable and field?

Reply:
Good idea. Now this is meant to tally up the total bytes that have been written but not compacted in the current lifetime of vector. But, the logic has got a little muddled and may not be correct any longer. Consider a fresh vector with an existing disk buffer. It starts up and `current_size` is 0 but as values are read out of the buffer their sizes are subtracted from `current_size`. If reads are more plentiful than writes this will result in a situation where the buffer is cleared out but not compacted and the writer is continually notified to wake up, inappropriately.

Reply:
Yes, buggy idea. I'll revert.

Reply:
👍 I was going to ask what other implications this might have. It seems like we could solve this, but it'll take a bit more work to rethink the compaction and writer notification strategy.

Reply:
Yep, I thought I had it worked out and then as I thought more about a new name I realized I didn't. :)